website/core/utils.py

```diff
@@ -74,26 +75,53 @@ def is_url(url):
 def load_data(path):
     # TODO replace with call to google sheets api directly
     # COLS: ['Type', 'Name', 'Contact', 'Url', 'Description', 'National', 'State', 'City']
+    # Opens file, reads it with csv package in csv format, extracts all data to variable 'rows',
+    # extracts header to variable cols, check if 'State' col is in 'cols'
+    # There are 4 different types: Business, Organization, Donate, Petition.
     with open(path) as f:
         reader = csv.DictReader(f, dialect='excel')
         rows = list(reader)
         cols = reader.fieldnames
         assert STATE_COL in cols
+    # state_codes will hold disctint values.
     state_codes = set()
     states_a = {}
+    # for every row in our csv file, add every distinct value into this set.
     for row in rows:
         state_codes.add(row[STATE_COL])
+    # for every state in the states_code array, create an array of their codes, skip over excluded values that are in our file and might have been inserted for States column.
     for code in sorted(state_codes):
         if code not in EXCLUDED_CODES:
             states_a[code] = ABBREV_TO_STATE[code.upper()]
     data = {}
+    national_data = {}
+    # for every row in our data, grab the 'Type' value, 'State' value, and 'National' value. Use these to check if they exists yet, if not initialize it.
+    # finally add data to that sub array in our data array.
     for row in rows:
         entry_type = row['Type']
         state = row['State']
+        national = row['National']
+        # print("National abbreviation ", national)
+        # print("state abbreviation ", state)
+
+        # initialize array
         if entry_type not in data:
             data[entry_type] = {}
+        # initialize sub-array
         if state not in data[entry_type]:
             data[entry_type][state] = []
-        # name, contact, url, desc, city
-        data[entry_type][state].append((row['Name'], row['Contact'], row['Url'], row['Description'], row['City']))
+            # print("initializing state array", state)
```

Comment:
Were the abbreviations changed for the national pages in the spreadsheet? Does it look right on your machine?

Reply:
No I just moved this row to 123 . Also lemme share with you the python script I have for pulling the info from the google sheets.